docs/nx-cloud/enterprise/dte/gitlab-dte.md

```diff
@@ -0,0 +1,75 @@
+# Custom Distributed Task Execution on GitLab
+
+Using [Nx Agents](/ci/features/distribute-task-execution) is the easiest way to distribute task execution, but it your organization may not be able to use hosted Nx Agents. With an [enterprise license](https://nx.app/enterprise), you can set up distributed task execution on your own CI provider using the recipe below.
+
+## Run Custom Agents on GitLab
+
+Run agents directly on GitLab with the workflow below:
+
+```yaml {% fileName=".gitlab-ci.yml" %}
+image: node:18
+
+# Creating template for DTE agents
+.dte-agent:
+  interruptible: true
+  cache:
+    key:
+      files:
+        - yarn.lock
+    paths:
+      - '.yarn-cache/'
+  script:
+    - yarn install --cache-folder .yarn-cache --prefer-offline --frozen-lockfile
+    - yarn nx-cloud start-agent
+
+# Creating template for a job running DTE (orchestrator)
+.base-pipeline:
+  interruptible: true
+  only:
+    - main
+    - merge_requests
+  cache:
+    key:
+      files:
+        - yarn.lock
+    paths:
+      - '.yarn-cache/'
+  before_script:
+    - yarn install --cache-folder .yarn-cache --prefer-offline --frozen-lockfile
+    - NX_HEAD=$CI_COMMIT_SHA
+    - NX_BASE=${CI_MERGE_REQUEST_DIFF_BASE_SHA:-$CI_COMMIT_BEFORE_SHA}
+  artifacts:
+    expire_in: 5 days
+    paths:
+      - dist
+
+# Main job running DTE
+nx-dte:
+  stage: affected
+  extends: .base-pipeline
+  script:
+    - yarn nx-cloud start-ci-run --stop-agents-after=build
+    - yarn nx-cloud record -- nx format:check --base=$NX_BASE --head=$NX_HEAD
+    - yarn nx affected --base=$NX_BASE --head=$NX_HEAD -t lint,test,build --parallel=2
```

Comment:
Same as above, add `e2e-ci` and the missing `NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: 3` env variable